kafka-client/src/main/java/dev/responsive/kafka/internal/db/rs3/RS3WindowFlushManager.java

```diff
@@ -0,0 +1,177 @@
+ * available at:
+ *
+ * https://www.responsive.dev/legal/responsive-bsl-10
+ *
+ * This software requires a valid Commercial License Key for production use. Trial and commercial
+ * licenses can be obtained at https://www.responsive.dev
+ */
+
+package dev.responsive.kafka.internal.db.rs3;
+
+import dev.responsive.kafka.internal.db.RemoteWriter;
+import dev.responsive.kafka.internal.db.WindowFlushManager;
+import dev.responsive.kafka.internal.db.partitioning.TablePartitioner;
+import dev.responsive.kafka.internal.db.rs3.client.LssId;
+import dev.responsive.kafka.internal.db.rs3.client.RS3Client;
+import dev.responsive.kafka.internal.db.rs3.client.WalEntry;
+import dev.responsive.kafka.internal.stores.RemoteWriteResult;
+import dev.responsive.kafka.internal.utils.WindowedKey;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+class RS3WindowFlushManager implements WindowFlushManager<Integer> {
+  private final UUID storeId;
+  private final RS3Client rs3Client;
+  private final LssId lssId;
+  private final RS3WindowTable table;
+  private final int kafkaPartition;
+  private final PssPartitioner pssPartitioner;
+  private final RS3WindowedKeySerde keySerde;
+  private final Map<Integer, Optional<Long>> writtenOffsets;
+
+  private long streamTime;
+  private final Map<Integer, Rs3WindowWriter> writers = new HashMap<>();
+
+  public RS3WindowFlushManager(
+      final UUID storeId,
+      final RS3Client rs3Client,
+      final LssId lssId,
+      final RS3WindowTable table,
+      final int kafkaPartition,
+      final PssPartitioner pssPartitioner,
+      final RS3WindowedKeySerde keySerde,
+      final Map<Integer, Optional<Long>> writtenOffsets,
+      final long initialStreamTime
+  ) {
+    this.storeId = storeId;
+    this.rs3Client = rs3Client;
+    this.lssId = lssId;
+    this.table = table;
+    this.kafkaPartition = kafkaPartition;
+    this.pssPartitioner = pssPartitioner;
+    this.keySerde = keySerde;
+    this.writtenOffsets = writtenOffsets;
+    this.streamTime = initialStreamTime;
+  }
+
+  @Override
+  public String tableName() {
+    return table.name();
+  }
+
+  @Override
+  public TablePartitioner<WindowedKey, Integer> partitioner() {
+    return new PssTablePartitioner<>(pssPartitioner) {
+      @Override
+      public byte[] serialize(final WindowedKey key) {
+        return keySerde.serialize(key);
+      }
+    };
+  }
+
+  @Override
+  public RemoteWriter<WindowedKey, Integer> createWriter(
+      final Integer pssId,
+      final long consumedOffset
+  ) {
+    final var writer = new Rs3WindowWriter(
+        storeId,
+        rs3Client,
+        table,
+        pssId,
+        lssId,
+        consumedOffset,
+        writtenOffsets.get(pssId),
+        kafkaPartition
+    );
+    writers.put(pssId, writer);
+    return writer;
+  }
+
+  @Override
+  public void writeAdded(final WindowedKey key) {
+    streamTime = Long.max(streamTime, key.windowStartMs);
```

Comment:
I know it's not actually being used anywhere at the moment so it doesn't matter too much, but to avoid confusion in the future we might want to name this more specifically to indicate that it isn't the persisted streamTime but rather the stream-time of the current pending batch (eg  `batchStreamTime` or `pendingStreamTime` or something better)

OTOH just thinking about how we'll want to use the streamTime in the future, maybe we actually do want the persisted streamTime only, to ensure it monotonically increases? Or will slateDB handle that for us automatically 🤔